File: client/src/app/layouts/training/model/trainee-attendance.ts
import { User } from 'src/app/views/pages/login/model/user.model';
import { BatchMaster } from './batch-master';
import { Trainee } from './trainee';

export class TraineeAttendance {
  trainingAttendanceId: number;
  batch: BatchMaster;
  session: String;
  trainee: Trainee;
  attendance: boolean;
  remark: string;
  createdBy: number;
  createdOn: Date;
  updatedBy: User;
  updatedOn: Date;
}

File: Java_Backend/src/main/java/com/netlink/rsk/model/TraineeAttendance.java
package com.netlink.rsk.model;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Data
@Entity
@Table(name = "Traineeattendance",catalog = "Training")
public class TraineeAttendance implements Serializable {

    @Id
    @Column(name = "Traineeattendanceid", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long trainingAttendanceId;

    @OneToOne
    @JoinColumn(name="Batchid",referencedColumnName = "Batchid")
    private BatchMaster batch;

    @Column(name = "Session")
    private String session;

    @OneToOne
    @JoinColumn(name = "Traineeid",referencedColumnName = "Traineeid")
    private Trainee trainee;

    @Column(name = "Attendance")
    private Boolean attendance;

    @Column(name = "Remark")
    private String remark;

    @Column(name = "Createdby")
    private Integer createdBy;

    @Column(name = "Createdon")
    private Date createdOn;

    @OneToOne
    @JoinColumn(name = "Updatedby", referencedColumnName = "Userid")
    private UserMaster updatedBy;

    @Column(name = "Updatedon")
    private Date updatedOn;
}
